Automattic/jetpack
# Republicize: add feature flag

~### Branched off from https://github.com/Automattic/jetpack/pull/21240~

This PR adds a feature flag as a way to deal with enabling/disabling the republicize feature.
The Republicize feature is not finished yet, so when enabled you'll see a toggle control but not too much else. This is going to be tackled in follow-up PRs.

#### Changes proposed in this Pull Request:

* Publicize: add republicize feature flag

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Test with Jetpack plugin activated
* Enable Sharing Jetpack module
* Create/edit a post
* Confirm anything has changed.
* Enable the feature flag through the filter
You can do it using the proper filter:

```php
add_filter( 'jetpack_block_editor_republicize_feature', '__return_true' );
```

Feature Flag disabled | Feature flag enabled | Feature enabled
------|------| ----
![image](https://user-images.githubusercontent.com/77539/136850324-a66dadb6-40d3-4461-a9d9-d82c42f83b49.png) | ![image](https://user-images.githubusercontent.com/77539/136850015-b305a48a-acf5-419c-a423-f60be29088f3.png) | ![image](https://user-images.githubusercontent.com/77539/136850440-588ab34d-3bf8-400f-a254-7f4cf68d3fb8.png)
